lib/yabeda/porta.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+# Based on https://github.com/yabeda-rb/yabeda-rails
+
+require "yabeda"
+
+module Yabeda
+  module Porta
+    LONG_RUNNING_REQUEST_BUCKETS = [
+      0.005, 0.01, 0.05, 0.1, 0.5, 1, 5, 10, # standard
+      30, 60, 600, # Sometimes requests may be really long-running
+    ].freeze
+
+    class << self
+      def controller_handlers
+        @controller_handlers ||= []
+      end
+
+      def on_controller_action(&block)
+        controller_handlers << block
+      end
+
+      def install!
+        Yabeda.configure do
+          group :rails
+
+          counter   :requests_total,   comment: "A counter of the total number of HTTP requests rails processed.",
+                    tags: %i[controller action status]
+
+          histogram :request_duration, tags: %i[controller action status],
+                    unit: :seconds,
+                    buckets: LONG_RUNNING_REQUEST_BUCKETS,
+                    comment: "A histogram of the response latency."
+
+          histogram :view_runtime, unit: :seconds, buckets: LONG_RUNNING_REQUEST_BUCKETS,
+                    comment: "A histogram of the view rendering time.",
+                    tags: %i[controller action status]
+
+          histogram :db_runtime, unit: :seconds, buckets: LONG_RUNNING_REQUEST_BUCKETS,
+                    comment: "A histogram of the activerecord execution time.",
+                    tags: %i[controller action status]
+
+          ActiveSupport::Notifications.subscribe "process_action.action_controller" do |*args|
+            event = ActiveSupport::Notifications::Event.new(*args)
+            labels = {
+              controller: event.payload[:params]["controller"],
+              action: event.payload[:params]["action"],
+              status: event.payload[:status],
+            }.merge!(event.payload.slice(*Yabeda.default_tags.keys))
+
+            rails_requests_total.increment(labels)
+            rails_request_duration.measure(labels, Yabeda::Porta.ms2s(event.duration))
+            rails_view_runtime.measure(labels, Yabeda::Porta.ms2s(event.payload[:view_runtime]))
+            rails_db_runtime.measure(labels, Yabeda::Porta.ms2s(event.payload[:db_runtime]))
+
+            Yabeda::Porta.controller_handlers.each do |handler|
+              handler.call(event, labels)
+            end
+          end
+        end
+      end
+
+      def ms2s(ms)
```

Comment:
another nitpick, to me more readable would be just `to_sec`. Once I figured out what this did, I see the method name very descriptive. But somehow had to look at the code to see what it did. Or maybe it's just too late... :)

I'm not sure as you see fit.

```suggestion
      def to_sec(ms)
```

Reply:
The suggestion is OK, but the thing is that I just copy-pasted this code from [yabeda-rails](https://github.com/yabeda-rb/yabeda-rails/blob/v0.8.1/lib/yabeda/rails.rb#L80)

But maybe it doesn't matter, we can just do our own thing and don't care about the initial implementation (as it has been refactored already since initially copy-pasted).